### PR TITLE
[BACKLOG-36242][BACKLOG-36243][BACKLOG-36244][BACKLOG-36254] 1) Fix N…

### PR DESCRIPTION
**Pull request by tkafalas**
…PE on invalid save attempt

2) Implement PasswordVisibleTextVar and replace all PasswordTextVar widets.
3) Add file browse to GCS dialog.
4) Add Rename/Copy prompt when saving VFS connection after changing the name.

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 55s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,plugins/connections/ui,ui
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/connections/ConnectionManager.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 package org.pentaho.di.connections;
```
Comment (peterrinehart) [inline core/src/main/java/org/pentaho/di/connections/ConnectionManager.java:23]: Update the copyright date on this file

File: plugins/connections/ui/src/main/java/org/pentaho/di/connections/ui/dialog/ConnectionRenameDialog.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2019 by Hitachi Vantara : http://www.pentaho.com
```
Comment (peterrinehart) [inline plugins/connections/ui/src/main/java/org/pentaho/di/connections/ui/dialog/ConnectionRenameDialog.java:5]: Update the date here

File: ui/src/main/java/org/pentaho/di/ui/core/widget/PasswordVisibleTextVar.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2017 by Hitachi Vantara : http://www.pentaho.com
```
Comment (peterrinehart) [inline ui/src/main/java/org/pentaho/di/ui/core/widget/PasswordVisibleTextVar.java:5]: Copyright date

**Review by peterrinehart — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 46s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,plugins/connections/ui,ui
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Review by ddiroma — APPROVED** [review]

**Pull request merged**